main.go

```diff
@@ -319,6 +341,16 @@ func mainerr() error {
 
 			gobin := filepath.Join(gobinCache, mainrel)
 
+			// If we have non-zero -tags then we need to hash the path
+			if *fTags != "" {
+				// TODO we could get smart about building a sorted, uniq list of build constraints
+				// but for now let's just be stupid
+				h := sha256.New()
+				fmt.Fprintln(h, mainrel)
+				fmt.Fprintln(h, *fTags)
```

Comment:
Given that the tag list can be specified in any order, perhaps it might be worth sorting the tag list before hashing?

Reply:
Yeh that was the essence of the comment above this. We could/should also simplify the boolean expression... but I went for quick dirty for now.